diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4445a951..4da6662b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,5 @@
 cmake_minimum_required(VERSION 3.6)
+cmake_policy(SET CMP0091 NEW)
 project(vgmstream C)
 enable_language(CXX)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 38e949b4..991f4f51 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -47,3 +47,27 @@ target_include_directories(libvgmstream PRIVATE
 	layout
 	meta
 	util)
+
+if(MSVC)
+	target_compile_definitions (libvgmstream PRIVATE "USE_ALLOCA")
+endif()
+
+# Install library
+install(TARGETS libvgmstream
+  RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+  ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+  LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+)
+
+if(NOT SKIP_INSTALL_HEADERS)
+	# Install headers
+	install( FILES ${CMAKE_CURRENT_SOURCE_DIR}/vgmstream.h DESTINATION include/libvgmstream )
+	install( FILES ${CMAKE_CURRENT_SOURCE_DIR}/plugins.h DESTINATION include/libvgmstream )
+	install( FILES ${CMAKE_CURRENT_SOURCE_DIR}/util.h DESTINATION include/libvgmstream )
+	install( FILES ${CMAKE_CURRENT_SOURCE_DIR}/util.h DESTINATION include/libvgmstream )
+	install( FILES ${CMAKE_CURRENT_SOURCE_DIR}/util/log.h DESTINATION include/libvgmstream/util )
+	install( FILES ${CMAKE_CURRENT_SOURCE_DIR}/streamtypes.h DESTINATION include/libvgmstream )
+	install( FILES ${CMAKE_CURRENT_SOURCE_DIR}/streamfile.h DESTINATION include/libvgmstream )
+	install( FILES ${CMAKE_CURRENT_SOURCE_DIR}/coding/g72x_state.h DESTINATION include/libvgmstream/coding )
+	install( FILES ${CMAKE_CURRENT_SOURCE_DIR}/coding/nwa_decoder.h DESTINATION include/libvgmstream/coding )
+endif()
